scripts/convert_to_tarred_audio_dataset.py

```diff
@@ -68,21 +72,44 @@ def create_shard(entries, target_dir, new_entries, shard_id):
     """
     tar = tarfile.open(os.path.join(target_dir, f'audio_{shard_id}.tar'), mode='w')
 
+    count = dict()
     for entry in entries:
         # We squash the filename since we do not preserve directory structure of audio files in the tarball.
         base, ext = os.path.splitext(entry['audio_filepath'])
         base = base.replace('/', '_')
         # Need the following replacement as long as WebDataset splits on first period
         base = base.replace('.', '_')
         squashed_filename = f'{base}{ext}'
-        tar.add(entry['audio_filepath'], arcname=squashed_filename)
-
-        new_entry = {
-            'audio_filepath': squashed_filename,
-            'duration': entry['duration'],
-            'text': entry['text'],
-            'shard_id': shard_id,  # Keep shard ID for recordkeeping
-        }
+        if squashed_filename not in count:
+            tar.add(entry['audio_filepath'], arcname=squashed_filename)
+
+        if entry['label']:
```

Comment:
What if `label`  key doesnt exist in entry - ASR manifests dont have label but text instead? First check for key, then access it.

Reply:
good catch!!!